ast/builtins.go

```diff
@@ -1667,6 +1670,20 @@ var CastObject = &Builtin{
 	),
 }
 
+// Lookup returns takes an object and returns a value under its key if present,
+// otherwise it returns the default.
+var Lookup = &Builtin{
+	Name: "lookup",
```

Comment:
Nit: let's namespace this and call it `object.get` instead. We discussed this a bit in #1617. Eventually, like @srenatus mentioned, we'll want to replace usage of this with an operator but for the time being it's fine to add the built-in function.

Reply:
I could give the operator a shot. In fact, do we need an actual operator, or would an infix built-in work? I'd need a small hint though - how do I go about detecting that the first operand is actually undefined - I couldn't figure it out thus far.

Reply:
Done.